core/src/epicli/data/common/ansible/playbooks/roles/sshtunnel/files/tunnel.sh

```diff
@@ -1,12 +1,48 @@
 #!/bin/bash
 
-set -o errexit
+set -o errexit -o nounset -o pipefail
+
+export AUTOSSH_PIDFILE=/tmp/autossh-kubectl-tunnel.pid
+
+is_port_available() {
+  local port=$1
+
+  # nc command is much faster than lsof and doesn't require sudo to take into account ports used by other users
+  nc -z 127.0.0.1 "$port" && return 1 || return 0 # 0 if port is available
+}
 
```

Comment:
Thanks to that a function can be used inside a subshell :+1: (assuming we are using BASH):
```suggestion

export -f is_port_available

```